packages/datetime/src/dateInput.tsx

```diff
@@ -337,9 +352,18 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
                 this.setState({ isInputFocused: false });
             }
         }
+        this.safeInvokeInputProp("onBlur", e);
     }
 
     private setInputRef = (el: HTMLElement) => {
         this.inputRef = el;
+        const { inputProps = {} } = this.props;
+        Utils.safeInvoke(inputProps.inputRef, el);
+    }
+
+    /** safe wrapper around invoking input props event handler (prop defaults to undefined) */
+    private safeInvokeInputProp(name: keyof HTMLInputProps, e: React.SyntheticEvent<HTMLElement>) {
+        const { inputProps = {} } = this.props;
+        Utils.safeInvoke(inputProps[name], e);
```

Comment:
thoughts on this pattern? it's a way to avoid defining `defaultProps.inputProps = {}` to prevent accidental mutation like in #980. but maybe it's just too much overhead and we should set the default prop and keep an eye out for potential mutation?